flink-doris-connector/src/main/java/org/apache/doris/flink/flight/DorisFlightSource.java

```diff
@@ -0,0 +1,206 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.flink.flight;
+
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.connector.source.Boundedness;
+import org.apache.flink.api.connector.source.Source;
+import org.apache.flink.api.connector.source.SourceReader;
+import org.apache.flink.api.connector.source.SourceReaderContext;
+import org.apache.flink.api.connector.source.SplitEnumerator;
+import org.apache.flink.api.connector.source.SplitEnumeratorContext;
+import org.apache.flink.api.java.typeutils.ResultTypeQueryable;
+import org.apache.flink.core.io.SimpleVersionedSerializer;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import org.apache.doris.flink.cfg.DorisOptions;
+import org.apache.doris.flink.cfg.DorisReadOptions;
+import org.apache.doris.flink.deserialization.DorisDeserializationSchema;
+import org.apache.doris.flink.flight.arrow.DorisFlightRecordEmitter;
+import org.apache.doris.flink.flight.arrow.DorisFlightSourceReader;
+import org.apache.doris.flink.flight.assigners.FlightSplitAssigner;
+import org.apache.doris.flink.flight.enumerator.DorisFlightSourceEnumerator;
+import org.apache.doris.flink.flight.enumerator.PendingFlightSplitsCheckpointSerializer;
+import org.apache.doris.flink.flight.split.DorisFlightSourceSplit;
+import org.apache.doris.flink.flight.split.DorisFlightSourceSplitSerializer;
+import org.apache.doris.flink.rest.PartitionDefinition;
+import org.apache.doris.flink.rest.RestService;
+import org.apache.doris.flink.source.enumerator.PendingSplitsCheckpoint;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/** DorisSource based on FLIP-27 which is a BOUNDED stream. */
```

Comment:
Modify the comments of the following class to explain the role of DorisFlightSource

Reply:
modified